### PR TITLE
Fix missing completions for field names in MappingConstructorExpressionNodeContext

### LINKED ISSUE #34822: Missing completions for field names in MappingConstructorExpressionNodeContext
**Description:**

Missing completions for the field names `name`, `age` in the scenario below.

Screenshot 2022-01-24 at 12 16 15

**Steps to reproduce:**
```
type Person record {
    readonly string name;
    int age;
};

type PersonTable table<Person> key(name);

public function main() {

PersonTable t1 = table [
    {name:"A", age:2},
    {na<cursor>}
];

}
```

**Affected Versions:**
Ballerina 2201.0.0 (Swan Lake) RC2

### PR DESCRIPTION
**Pull request by gayalkuruppu**
Fixes #34822 

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/util/ContextTypeResolver.java [hunk 1]
```diff
@@ -643,6 +643,11 @@ public ContextTypeResolver(PositionedOperationContext context) {
 
     @Override
     public Optional<TypeSymbol> transform(MappingConstructorExpressionNode mappingConstructorExpressionNode) {
+        Optional<TypeSymbol> typeSymbol = context.currentSemanticModel()
+                .flatMap(semanticModel -> semanticModel.typeOf(mappingConstructorExpressionNode));
+        if (typeSymbol.isPresent() && typeSymbol.get().typeKind() != TypeDescKind.COMPILATION_ERROR) {
+            return typeSymbol;
+        }
```
Comment (IMS94) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/util/ContextTypeResolver.java:650]: We can rewrite this part as below:
```
return context.currentSemanticModel()
                .flatMap(semanticModel -> semanticModel.typeOf(mappingConstructorExpressionNode))
                .filter(typeSymbol -> typeSymbol.typeKind() != TypeDescKind.COMPILATION_ERROR)
                .or(() -> mappingConstructorExpressionNode.parent().apply(this));
```
Btw, I have done the same change in #34958. Depending on which PR gets merged first, let's rebase.

**Review by IMS94 — CHANGES_REQUESTED** [review]
LGTM! There's a minor comment on how we can rewrite a logic.

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/util/ContextTypeResolver.java (hunk 1 above)
Comment (gayalkuruppu) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/util/ContextTypeResolver.java:650]: If rebasing, itll leave the PR only with a test case. In that case, do we still need the PR?

Comment (IMS94) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/util/ContextTypeResolver.java:650]: We need the test case for sure :smile: So, we need the PR.

**Comment by mohanvive** [comment]
@gayalkuruppu shall we rebase this please?

**Comment by gayalkuruppu** [comment]
> @gayalkuruppu shall we rebase this please?

Done

**Review by mohanvive — APPROVED** [review]

**Comment by mohanvive** [comment]
@gayalkuruppu Can you please send this to patch branch?

**Review by nadeeshaan — APPROVED** [review]
LGTM

**Review by mohanvive — APPROVED** [review]

**Pull request merged**